core-rust/state-manager/src/types.rs

```diff
@@ -663,6 +667,38 @@ pub struct PrepareGenesisRequest {
 
 #[derive(Debug, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
 pub struct PrepareGenesisResult {
-    pub validator_set: Option<BTreeMap<ComponentAddress, Validator>>,
+    pub validator_set: Option<Vec<ActiveValidatorInfo>>,
     pub ledger_hashes: LedgerHashes,
 }
+
+#[derive(Debug, Clone, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
+pub struct TimestampedValidatorSignature {
+    pub key: EcdsaSecp256k1PublicKey,
+    pub validator_address: Option<ComponentAddress>,
+    pub timestamp_ms: i64,
```

Comment:
how about making it an unsigned int? we could then add some assertions when converting to `long` on the Java side (or simply use `UInt64`).

same for other timestamps below

Reply:
I just tried and propagated that change until I hit this:
https://github.com/radixdlt/babylon-node/blob/48b2d2116750e8812e726ef3f80aee3b179364b9/core-rust/state-manager/src/transaction/validator_transaction.rs#L43
(the `ClockSetCurrentTimeInput.current_time_ms` is i64)

Seems like our convention is to have millis in i64.

One fun argument I've heard:
- with u64, you can only represent dates from year 1970 to year 478000000
- while with i64, you can represent dates from year -239000000 to 239000000.
And people choose the latter, since some domains care about retro-populated events that happened before 1970.

The `i64` isn't really limiting anyone in foreseeable future, and a negative value is perfectly interpretable (so, not illegal - for our domain `-1000` is most likely invalid, but `1000` is then invalid too).